Frameworks/CoreGraphics/DWriteFontBinaryDataLoader.mm

```diff
@@ -0,0 +1,120 @@
+
+        // Concat filetime into a single uint64_t
+        m_lastWriteTime = 0;
+        m_lastWriteTime |= static_cast<uint64_t>(fileTime.dwLowDateTime);
+        m_lastWriteTime |= static_cast<uint64_t>(fileTime.dwHighDateTime) << 32;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetFileSize(_Out_ uint64_t* fileSize) {
+        RETURN_HR_IF_NULL(E_POINTER, fileSize);
+        *fileSize = CFDataGetLength(m_data.get());
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE GetLastWriteTime(_Out_ uint64_t* lastWriteTime) {
+        RETURN_HR_IF_NULL(E_POINTER, lastWriteTime);
+        *lastWriteTime = m_lastWriteTime;
+        return S_OK;
+    };
+
+    // fragmentContext is deliberately unused: this is meant to be passed to ReleaseFileFragment() below to free part of the data,
+    // but this stream frees all the underlying CFData at once
+    HRESULT STDMETHODCALLTYPE ReadFileFragment(_Out_ const void** fragmentStart,
+                                               uint64_t fileOffset,
+                                               uint64_t fragmentSize,
+                                               _Out_ void** fragmentContext) {
+        if (fileOffset + fragmentSize > CFDataGetLength(m_data.get())) {
+            return E_INVALIDARG;
+        }
+
+        if (fragmentStart) {
+            const uint8_t* underlyingBuffer = CFDataGetBytePtr(m_data.get());
+            *fragmentStart = reinterpret_cast<const void*>(&(underlyingBuffer[fileOffset]));
+        }
+
+        if (fragmentContext) {
+            *fragmentContext = nullptr;
+        }
+
+        return S_OK;
+    };
+
+    void STDMETHODCALLTYPE ReleaseFileFragment(void* fragmentContext){
+        // Deliberate no-op: data is released alongside the m_data member with the destruction of this object
+    };
+
+private:
+    woc::unique_cf<CFDataRef> m_data;
+    uint64_t m_lastWriteTime;
+};
+
+// DWriteFontBinaryDataLoader member functions
+
+DWriteFontBinaryDataLoader::DWriteFontBinaryDataLoader() {
+    // Deliberate no-op
+}
+
+HRESULT DWriteFontBinaryDataLoader::RuntimeClassInitialize(CFDataRef data) {
+    RETURN_HR_IF_NULL(E_INVALIDARG, data);
+
+    CFRetain(data);
+    m_data.reset(data);
+    return S_OK;
+}
+
+// Ignores first two params, just return the same kind of stream always
+// Stream returned is dictated by dataProvider passed in at initialization time
+HRESULT STDMETHODCALLTYPE DWriteFontBinaryDataLoader::CreateStreamFromKey(_In_ const void* fontFileReferenceKey,
+                                                                          uint32_t fontFileReferenceKeySize,
+                                                                          _Out_ IDWriteFontFileStream** fontFileStream) {
+    RETURN_HR_IF_NULL(E_POINTER, fontFileStream);
+
+    ComPtr<DWriteFontBinaryDataStream> ret;
+    RETURN_IF_FAILED(MakeAndInitialize<DWriteFontBinaryDataStream>(&ret, m_data.get()));
+
+    return ret.CopyTo(fontFileStream);
```

Comment:
>CopyTo [](start = 15, length = 6)

prefer detach here if you aren't saving the reference internally for later. CopyTo adds an extra AddRef that technically isn't needed since you are *transferring* ownership.